jetbrains-ultimate/tst-201/software/aws/toolkits/jetbrains/services/redshift/auth/RedshiftAwsAuthTest.kt

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.redshift.auth
+
+import com.intellij.database.dataSource.DatabaseConnectionInterceptor.ProtoConnection
+import com.intellij.database.dataSource.DatabaseConnectionPoint
+import com.intellij.database.dataSource.LocalDataSource
+import com.intellij.testFramework.ProjectRule
+import com.nhaarman.mockitokotlin2.KArgumentCaptor
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.doAnswer
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.doThrow
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.stub
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
+import software.amazon.awssdk.services.redshift.RedshiftClient
+import software.amazon.awssdk.services.redshift.model.Cluster
+import software.amazon.awssdk.services.redshift.model.DescribeClustersRequest
+import software.amazon.awssdk.services.redshift.model.DescribeClustersResponse
+import software.amazon.awssdk.services.redshift.model.GetClusterCredentialsRequest
+import software.amazon.awssdk.services.redshift.model.GetClusterCredentialsResponse
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.core.utils.RuleUtils
+import software.aws.toolkits.jetbrains.core.MockClientManagerRule
+import software.aws.toolkits.jetbrains.core.credentials.ConnectionSettings
+import software.aws.toolkits.jetbrains.core.credentials.MockCredentialsManager
+import software.aws.toolkits.jetbrains.core.datagrip.CREDENTIAL_ID_PROPERTY
+import software.aws.toolkits.jetbrains.core.datagrip.REGION_ID_PROPERTY
+import software.aws.toolkits.jetbrains.core.region.MockRegionProvider
+
+class RedshiftAwsAuthTest {
+    @Rule
+    @JvmField
+    val projectRule = ProjectRule()
+
+    @Rule
+    @JvmField
+    val mockClientManager = MockClientManagerRule { projectRule.project }
+
+    private val mockCreds = AwsBasicCredentials.create("Access", "ItsASecret")
+
+    private val apiAuth = AwsAuth()
+    private val credentialId = RuleUtils.randomName()
+    private val defaultRegion = RuleUtils.randomName()
+    private val region = AwsRegion(defaultRegion, RuleUtils.randomName(), RuleUtils.randomName())
+    private val clusterId = RuleUtils.randomName()
+    private val username = RuleUtils.randomName()
+
+    private val redshiftSettings = RedshiftSettings(
+        clusterId = clusterId,
+        username = username,
+        connectionSettings = ConnectionSettings(mock(), region)
+    )
+
+    @Before
+    fun setUp() {
+        MockCredentialsManager.getInstance().addCredentials(credentialId, mockCreds)
```

Comment:
If you dont care about the value of the creds, you can commit the parameter

Reply:
It's easier to use it IMO. We don't validate it but we do have to return the ID so I think it makes more sense to make one than to reach into the default mock credential.